_posts/2022-02-08-opensearch-java-runtime.markdown

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title: "Using different Java runtimes with OpenSearch"
+authors:
+  - dblock
+  - smortex
+  - reta
+  - ryanbogan
+date: 2022-02-08
+categories:
+  - technical-post
+twittercard:
+  description: "OpenSearch ships with a bundled Java Development Kit (JDK) that has recently been updated to version 17 (LTS). In this blog post we'll explain this change, and describe new features that make swapping the JDK easier."
+excerpt: ""
+
+---
+At the time of the fork OpenSearch inherited bundling OpenJDK 15, and we have made 8 releases with AdoptOpenJDK 15.0.1+9 as the default runtime, replaced with Adoptium (Temurin) 17.0.2+8 in OpenSearch 1.3.0, which is a Long-Term Support (LTS) release. LTS releases focus on stability, therefore you can expect future versions of OpenSearch to always default to bundling an LTS JDK.
+
+| Version                                                         | Bundled JDK (Linux)         |
+|:---------------------------------------------------------------:|:---------------------------:|
+| [1.0.0](https://opensearch.org/versions/opensearch-1-0-0.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.0.1](https://opensearch.org/versions/opensearch-1-0-1.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.1.0](https://opensearch.org/versions/opensearch-1-1-0.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.2.0](https://opensearch.org/versions/opensearch-1-2-0.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.2.1](https://opensearch.org/versions/opensearch-1-2-1.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.2.2](https://opensearch.org/versions/opensearch-1-2-2.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.2.3](https://opensearch.org/versions/opensearch-1-2-3.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.2.4](https://opensearch.org/versions/opensearch-1-2-4.html)  | AdoptOpenJDK 15.0.1+9       |
+| [1.3.0](https://opensearch.org/versions/opensearch-1-3-0.html)  | Adoptium (Temurin) 17.0.2+8 |
+
+This, however, doesn't tell the whole story. The OpenSearch distribution is comprised of the engine and a dozen plugins. Both the engine and each plugin is built and tested with a range of JDKs (a subset of 8, 11, 14, 15, 17, etc.), across multiple operating systems (Linux, Windows, FreeBSD, MacOS, etc.). Finally, the complete distribution is rebuilt from source, and tested with the bundled JDK.
```

Comment:
>  a range of JDKs (a subset of 8, 11, 14, 15, 17, etc.)

I think readers want to know that OpenSearch is tested against LTS releases above all else. You might be able to say:

> a range of JDKs, including LTS releases